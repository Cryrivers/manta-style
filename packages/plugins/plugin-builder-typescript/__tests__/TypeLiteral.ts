import getTranspiledString from '../src/utils/transpile';

describe('Type Literal', () => {
  test('Basic properties', () => {
    const source = `
        type Test = {
            a: number,
            b: 1,
            c?: string,
            d: string[]
        }
      `;
    expect(getTranspiledString(source)).toMatchSnapshot();
  });
});
