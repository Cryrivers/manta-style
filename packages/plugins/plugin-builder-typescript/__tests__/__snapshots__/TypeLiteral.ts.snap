// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type Literal Basic properties 1`] = `
"\\"use strict\\";
exports.__esModule = true;
var runtime_1 = require(\\"@manta-style/runtime\\");
var Test = runtime_1[\\"default\\"].TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
    var type = runtime_1[\\"default\\"].TypeLiteral(function (typeLiteral) {
        typeLiteral.property(\\"a\\", runtime_1[\\"default\\"].NumberKeyword, false, []);
        typeLiteral.property(\\"b\\", runtime_1[\\"default\\"].Literal(1), false, []);
        typeLiteral.property(\\"c\\", runtime_1[\\"default\\"].StringKeyword, true, []);
        typeLiteral.property(\\"d\\", runtime_1[\\"default\\"].ArrayType(runtime_1[\\"default\\"].StringKeyword), false, []);
    });
    return type;
}, []);
"
`;
